### ISSUE TITLE
Issue with ending interactive Run with Run.complete()

### ISSUE THREAD

**Issue by swanderz** [Azure/MachineLearningNotebooks#1471]
| entity        | `id` |
|:------------------|:---------------------------------------|
| `run`          | `827937b4-4bc4-49e3-8307-4997de58bc3f` |
| `subscription` | `ca4b8090-9bfc-4c02-a5bb-78871d8c8d1a` |

I'm using the combo of `Experiment.start_logging()` and `Run.complete()` within a Python script to simulate submission to a remote compute.

Questions:
- What is being uploaded? In the workspace UI, I can see that the metrics, `outputs` file and the `source_directory` snapshot upload successfully, but the run doesn't complete.
- Am I wrong to assume this works outside of a Compute Instance? (It'd be incredible if it did!!!)
- What auth is AML using here? I'm using InteractiveAuth for the pipeline, but I'm also authenticated with the `azure cli`

### Error message
>  Encountered authorization error while uploading to blob storage. Please check the storage account attached to your workspace. Make sure that the current user is authorized to access the storage account and that the request is not blocked by a firewall, virtual network, or other security setting.

### (almost) Minimum Reproducible Error

Basically the usage is this:
```python
from azureml.core import Experiment, Run, Workspace

interactive_auth = InteractiveLoginAuthentication(tenant_id='muh_tenant')
ws = Workspace.get('blahblah', auth=interactive_auth)
exp = Experiment(ws, 'KingCody')

run = exp.start_logging(outputs='where/data/is_saved.csv', snapshot_directory='.')

# good stuff in here
df = heckin_good_code()
df.to_csv('where/data/is_saved.csv')

run.complete()
```

### Full Stacktrace

```python
Traceback (most recent call last):
  File "src/oneshot/oneshot.py", line 552, in <module>
    run.complete()
  File "/Users/anders.swanson/opt/miniconda3/envs/rvo/lib/python3.8/site-packages/azureml/core/run.py", line 1342, in complete
    self._client.complete(_set_status=_set_status)
  File "/Users/anders.swanson/opt/miniconda3/envs/rvo/lib/python3.8/site-packages/azureml/_run_impl/run_history_facade.py", line 594, in complete
    self.upload_tracked_files()
  File "/Users/anders.swanson/opt/miniconda3/envs/rvo/lib/python3.8/site-packages/azureml/_run_impl/run_history_facade.py", line 660, in upload_tracked_files
    self.output_file_tracker.upload_tracked_files()
  File "/Users/anders.swanson/opt/miniconda3/envs/rvo/lib/python3.8/site-packages/azureml/_history/utils/context_managers.py", line 153, in upload_tracked_files
    self.py_wd.track(self.artifacts_client, self.container_id, self.trackfolders, self.deny_list)
  File "/Users/anders.swanson/opt/miniconda3/envs/rvo/lib/python3.8/site-packages/azureml/_history/utils/context_managers.py", line 95, in track
    fs.track(artifacts_client, container_id, track_folders, blacklist)
  File "/Users/anders.swanson/opt/miniconda3/envs/rvo/lib/python3.8/site-packages/azureml/_history/utils/filesystem.py", line 81, in track
    raise AzureMLAggregatedException(exception_messages)
azureml._common.exceptions.AzureMLAggregatedException: AzureMLAggregatedException:
        Message: Encountered authorization error while uploading to blob storage. Please check the storage account attached to your workspace. Make sure that the current user is authorized to access the storage account and that the request is not blocked by a firewall, virtual network, or other security setting.
        StorageAccount: sasetfdevdsresvalue
        ContainerName: azureml
        StatusCode: 403
        InnerException None
        ErrorResponse 
{
    "error": {
        "message": "Encountered authorization error while uploading to blob storage. Please check the storage account attached to your workspace. Make sure that the current user is authorized to access the storage account and that the request is not blocked by a firewall, virtual network, or other security setting.\n\tStorageAccount: sasetfdevdsresvalue\n\tContainerName: azureml\n\tStatusCode: 403"
    }
}
```

**Comment by nishankgu** [comment]
Could you clarify how you are submitting your job? Is this a local run on a compute isntance, or a remote run? The title of the issue says interactive run, so wanted to double check here.

**Comment by swanderz** [comment]
@nishankgu, I have this code in `oneshot.py` that I am executing on my local machine from the terminal with `python oneshot.py`. it is only interactive in the sense that it uses `InteractiveAuthentication` so that when I submit the script, a browser window pops up asking me to login.

**Comment by swanderz** [comment]
Clarifying a question I've gotten on this. I also edited my original question to include a `run_id`.

> What is being uploaded? In the workspace UI, I can see that the metrics, outputs file and the source_directory snapshot upload successfully, but the run doesn't complete.

The error message says "error uploading to blob storage". This made me think that the function failed to upload the snapshot data and outputs to the Run history in blob. However, when I check the created run in the Azure ML UI, I can see see contents of the local `snapshot_directory` were uploaded to the `Run`, and the same is the case with the files I saved locally, the paths of which I then passed to the argument: `outputs`.
